ocean_lib/models/v4/test/test_vesting.py

```diff
@@ -0,0 +1,198 @@
+from ocean_lib.models.v4.erc721_factory import ERC721FactoryContract
+from ocean_lib.models.v4.erc721_token import ERC721Token
+from ocean_lib.models.v4.models_structures import ErcCreateData, PoolData
+from ocean_lib.models.v4.side_staking import SideStaking
+from ocean_lib.web3_internal.constants import ZERO_ADDRESS
+from tests.resources.helper_functions import get_address_of_type
+
+
+def test_main(
+    web3,
+    config,
+    consumer_wallet,
+    publisher_wallet,
+    another_consumer_wallet,
+    factory_router,
+):
+    """main test flow"""
+
+    vesting_amount = web3.toWei("0.0018", "ether")
+
+    nftFactory = ERC721FactoryContract(
+        web3=web3, address=get_address_of_type(config, "ERC721Factory")
+    )
+
+    erc721_factory = ERC721FactoryContract(
+        web3,
+        get_address_of_type(config, "ERC721Factory"),
+    )
+
+    """test deploy erc721"""
+    tx = erc721_factory.deploy_erc721_contract(
+        "NFT",
+        "NFTS",
+        1,
+        ZERO_ADDRESS,
+        "https://oceanprotocol.com/nft/",
+        publisher_wallet,
+    )
+    tx_receipt = web3.eth.wait_for_transaction_receipt(tx)
+    registered_event = erc721_factory.get_event_log(
+        ERC721FactoryContract.EVENT_NFT_CREATED,
+        tx_receipt.blockNumber,
+        web3.eth.block_number,
+        None,
+    )
+    assert registered_event[0].event == "NFTCreated"
+    assert registered_event[0].args.admin == publisher_wallet.address
+    erc721_contract = ERC721Token(
+        web3=web3, address=registered_event[0].args.newTokenAddress
+    )
+
+    symbol = erc721_contract.symbol()
+    assert symbol == "NFTS"
+
+    ownerBalance = erc721_contract.contract.caller.balanceOf(publisher_wallet.address)
+    assert ownerBalance == 1
+    assert erc721_factory.get_nft_template(0)
+
+    """test roles"""
+    erc721_contract.add_manager(another_consumer_wallet.address, publisher_wallet)
+
+    erc721_contract.add_to_create_erc20_list(
+        consumer_wallet.address, another_consumer_wallet
+    )
+    erc721_contract.add_to_metadata_list(
+        consumer_wallet.address, another_consumer_wallet
+    )
+    erc721_contract.add_to_725_store_list(
+        consumer_wallet.address, another_consumer_wallet
+    )
+
+    permissions = erc721_contract.get_permissions(consumer_wallet.address)
+
+    assert permissions[1] == True
+    assert permissions[2] == True
+    assert permissions[3] == True
+
+    """test user 3 deploys an ERC20DT"""
+    ercData = ErcCreateData(
+        1,
+        ["ERC20DT1", "ERC20DT1Symbol"],
+        [
+            consumer_wallet.address,
+            another_consumer_wallet.address,
+            publisher_wallet.address,
+            ZERO_ADDRESS,
+        ],
+        [web3.toWei(0.05, "ether"), 0],
+        [b""],
+    )
+
+    trxErc20 = erc721_contract.create_erc20(ercData, consumer_wallet)
+
+    receipt = web3.eth.wait_for_transaction_receipt(trxErc20)
+    assert receipt["status"] == 1
```

Comment:
Try to access the status with `tx_receipt.status` since tx_receipt is an AttributeDict.